lib/streamlit/components/v1/custom_component.py

```diff
@@ -205,23 +205,17 @@ def deserialize_component(ui_value, widget_id=""):
                 widget_value = default
             elif isinstance(widget_value, ArrowTableProto):
                 widget_value = component_arrow.arrow_proto_to_dataframe(widget_value)
-
-            # widget_value will be either None or whatever the component's most
-            # recent setWidgetValue value is. We coerce None -> NoValue,
-            # because that's what DeltaGenerator._enqueue expects.
-            return widget_value if widget_value is not None else NoValue
+            return widget_value
 
         # We currently only support writing to st._main, but this will change
         # when we settle on an improved API in a post-layout world.
         dg = _main
 
         element = Element()
         return_value = marshall_component(dg, element)
-        result = dg._enqueue(
-            "component_instance", element.component_instance, return_value
-        )
 
-        return result
```

Comment:
Do you see a chance that we break existing custom components similar to what happened in https://github.com/streamlit/streamlit/pull/8193 when we changed the import/export module structure

Reply:
Nope, I don't think so since this is purely internal logic and I expect it to behave exactly as before.